api/agent/agent.go

```diff
@@ -690,7 +700,11 @@ func (a *agent) runHot(ctxArg context.Context, call *call, tok ResourceToken) {
 
 			// wait for this call to finish
 			// NOTE do NOT select with shutdown / other channels. slot handles this.
-			<-done
+			err := <-done
```

Comment:
this seems less than ideal, if a user's task times out then it causes the container to shut down. we don't need to do this, we just need to strip the pipes away from them -- right now, we're sharing pipes between tasks. if we shut down every timeout we increase container churn a lot even though it's not really necessary. actually started on fixing this yday, have a test that fusses now and was gonna fix it up.

Reply:
so, the pipe swap will not fix this. If the container has a process that has failed, we can no longer pass new requests to it. Some examples; a broken function that continues to spit out output or reads partial input. A function that sleeps, etc. Similar example as a broken http req/resp, once the req/resp structure is broken, we can no longer issue requests through the same connection.

Reply:
this presumes the container's process will never get back to resuming correct operation. with deadline now, I think there's some hope that in the timeout case this isn't true (this only covers the sleep case if used properly).

the incorrect input/output spitting out is the case I was gunning for -- for this, the trade off is what's the likelihood that making another container of that image will simply do the same thing (i.e. not just a rare funny state)? it seems rather high, I think, and makes sense to not go about making a lot of them and instead keeping just one that goofs and errors. the goal here was to make this situation easier to debug, to be exact, right now it's not very clear.

if a container responds beyond the boundary with a response for a prior call, I can think of a couple things that might happen (need to test to see which). It seems like ideally... that write to stdout gets ignored, it resumes reading stdin and hops to the next task? 

gonna dig around on this, I think there's a possibility to recover at least in prominent cases.